alf/algorithms/algorithm.py

```diff
@@ -976,16 +986,29 @@ def update_with_gradient(self, loss_info, valid_masks=None, weight=1.0):
                 (``shape=(T, B), dtype=tf.float32``)
             weight (float): weight for this batch. Loss will be multiplied with
                 this weight before calculating gradient.
-
+            batch_info (BatchInfo): information about this batch returned by
+                ``ReplayBuffer.get_batch()``
         Returns:
             tuple:
             - loss_info (LossInfo): loss information.
             - params (list[(name, Parameter)]): list of parameters being updated.
         """
+        masks = None
+        if (batch_info is not None and batch_info.importance_weights != ()
+                and self._config.priority_replay_beta != 0):
+            masks = batch_info.importance_weights.pow(
```

Comment:
If mini_batch_length>1, then batch_info contains different importance weights along T. However, when replayed using the segment tree, the T timesteps  are sampled altogether using just the weight of the first step? Is it better to propagate the weight of the first step to the remaining T-1 steps?

Reply:
The shape of importance with is [B]